### ISSUE TITLE
key modifier with expression as handler breaks build

### ISSUE THREAD

**Issue by jonaskuske** [vuejs/vue-cli#3496]
### Version
3.4.1

### Reproduction link
[https://github.com/jonaskuske/vue-cli-issue-modifier-expression](https://github.com/jonaskuske/vue-cli-issue-modifier-expression)

### Environment info
```
Environment Info:

  System:
    OS: Linux 4.4 Ubuntu 16.04.5 LTS (Xenial Xerus)
    CPU: (8) x64 Intel(R) Core(TM) i7-4790 CPU @ 3.60GHz
  Binaries:
    Node: 10.13.0 - /usr/bin/node
    Yarn: 1.13.0 - /usr/bin/yarn
    npm: 6.8.0 - ~/.npm-global/bin/npm
  Browsers:
    Chrome: Not Found
    Firefox: Not Found
  npmPackages:
    @vue/babel-helper-vue-jsx-merge-props:  1.0.0-beta.2
    @vue/babel-plugin-transform-vue-jsx:  1.0.0-beta.2
    @vue/babel-preset-app:  3.4.1
    @vue/babel-preset-jsx:  1.0.0-beta.2
    @vue/babel-sugar-functional-vue:  1.0.0-beta.2
    @vue/babel-sugar-inject-h:  1.0.0-beta.2
    @vue/babel-sugar-v-model:  1.0.0-beta.2
    @vue/babel-sugar-v-on:  1.0.0-beta.2
    @vue/cli-overlay:  3.4.1
    @vue/cli-plugin-babel: ^3.4.0 => 3.4.1
    @vue/cli-service: ^3.4.0 => 3.4.1
    @vue/cli-shared-utils:  3.4.1
    @vue/component-compiler-utils:  2.6.0
    @vue/preload-webpack-plugin:  1.1.0
    @vue/web-component-wrapper:  1.2.0
    vue: ^2.6.6 => 2.6.7
    vue-hot-reload-api:  2.3.2
    vue-loader:  15.6.4
    vue-style-loader:  4.1.2
    vue-template-compiler: ^2.5.21 => 2.6.7
    vue-template-es2015-compiler:  1.9.0
  npmGlobalPackages:
    @vue/cli: 3.4.1
```

### Steps to reproduce
1. Use a key modifier that inserts a conditional into the resulting event handler function, like `.exact` or `.shift` 
2. Use an expression as handler, e.g. ` @click.exact="$emit && $emit('click')"`
2. Build the project, doesn't matter if for development, production or testing
3. Failed to compile.

### What is expected?
The project compiles and runs.

### What is actually happening?
The compiled render function has a Syntax Error and the project fails to compile.

---
<br>
The template

```html
<div @click.exact="$emit && $emit('✅')">Modifier bug</div>
```

results in the following error:

<img src="https://data.jonaskuske.com/compileerror.png" width="100%">

If we prettify the compiled function it looks like this:

```javascript
var render = function() {
  var _vm = this;
  var _h = _vm.$createElement;
  var _c = _vm._self._c || _h;
  
  return _c(
    'div',
    {
      on: {
        "click": function($event) {
          if ($event.ctrlKey || $event.shiftKey || $event.altKey || $event.metaKey) {
            return null;
            _vm.$emit && _vm.$emit('✅')
          }
        }
      // ! WE'RE MISSING A CLOSING BRACKET HERE!
    }, [_vm._v("Modifier bug")])
}
```
As I highlighted in the code, the Syntax is invalid because a closing bracket is missing. But the *actual* issue is the botched if-statement that's causing this.

The following
```js
if ($event.ctrlKey || $event.shiftKey || $event.altKey || $event.metaKey) {
  return null;
  _vm.$emit && _vm.$emit('✅')
}
```
needs to be
```js
if ($event.ctrlKey || $event.shiftKey || $event.altKey || $event.metaKey) return null
_vm.$emit && _vm.$emit('✅')
```
otherwise the user code will never run...  
<br>

Note that this issue **doesn't affect Vue's runtime compiler**, you can copy-paste
```javascript
import Vue from 'https://unpkg.com/vue/dist/vue.esm.browser.js'
new Vue({
  el: document.body.appendChild(document.createElement('div')),
  template: `<p @click.shift="$emit && $emit('click')">✅</p>`
})
```
into a Codepen and it works without a hitch.
This is also the reason why I filed the issue here and not in the repository for Vue itself, hope this was right. (or is this an issue with Vue Loader?)

**Comment by jkzing** [comment]
Thanks for reporting, this is fixed in `vue-template-es2015-compiler@1.9.1`, see details here: https://github.com/vuejs/vue-template-es2015-compiler/releases/tag/v1.9.1.

**Comment by jonaskuske** [comment]
Ah, perfect!  
Didn't know the ES2015 compiler is a separate package, that's why I didn't find the already existing issue 😅
